lib/solveEquation/stepThrough.js

```diff
@@ -67,6 +71,27 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
       return solveConstantEquation(equation, debug, steps);
     }
 
+    if (checks.canFindRoots(equation)) {
+      const [solutions, symbol] = getSolutionsAndSymbol(equation);
+
+      // Returning a single solution for now.
+      const roots = new Equation(symbol, solutions[0], '=');
+      steps.push(new EquationStatus(
+        ChangeTypes.FIND_ROOTS, equation, roots));
+
+      // Return because we are done at this point
+      return steps;
+    }
+    else if (checks.canFindRoots(originalEquation)) {
```

Comment:
Why not just do this check at the beginning instead of cloning `originalEquation`?

Also why can't you only do it there? why do you also have to do it after other updates? Generally we only make one change at a time unless we have substeps, if it doesn't catch it this time it loops, it'll catch it the next time, right?